admin/server/organizations.go

```diff
@@ -301,63 +314,66 @@ func (s *Server) UpdateBillingSubscription(ctx context.Context, req *adminv1.Upd
-			}
-		} else {
-			// multiple subscriptions, cancel them first immediately and assign new plan should not happen unless externally assigned multiple subscriptions to the same org in the billing system.
-			// RepairOrgBilling does not fix multiple subscription issue, we are not sure which subscription to cancel and which to keep. However, in case of plan change we can safely cancel all older subscriptions and create a new one with new plan.
-			for _, sub := range subs {
-				err = s.admin.Biller.CancelSubscription(ctx, sub.ID, billing.SubscriptionCancellationOptionImmediate)
-				if err != nil {
-					return nil, status.Error(codes.Internal, err.Error())
-				}
-			}
-
-			// create new subscription
-			_, err = s.admin.Biller.CreateSubscription(ctx, org.BillingCustomerID, plan)
+	} else {
+		// multiple subscriptions, cancel them first immediately and assign new plan should not happen unless externally assigned multiple subscriptions to the same org in the billing system.
+		// RepairOrgBilling does not fix multiple subscription issue, we are not sure which subscription to cancel and which to keep. However, in case of plan change we can safely cancel all older subscriptions and create a new one with new plan.
+		for _, sub := range subs {
+			err = s.admin.Biller.CancelSubscription(ctx, sub.ID, billing.SubscriptionCancellationOptionImmediate)
 			if err != nil {
 				return nil, status.Error(codes.Internal, err.Error())
 			}
 		}
 
-		org, err = s.admin.DB.UpdateOrganization(ctx, org.ID, &database.UpdateOrganizationOptions{
-			Name:                                org.Name,
-			Description:                         org.Description,
-			QuotaProjects:                       valOrDefault(plan.Quotas.NumProjects, org.QuotaProjects),
-			QuotaDeployments:                    valOrDefault(plan.Quotas.NumDeployments, org.QuotaDeployments),
-			QuotaSlotsTotal:                     valOrDefault(plan.Quotas.NumSlotsTotal, org.QuotaSlotsTotal),
-			QuotaSlotsPerDeployment:             valOrDefault(plan.Quotas.NumSlotsPerDeployment, org.QuotaSlotsPerDeployment),
-			QuotaOutstandingInvites:             valOrDefault(plan.Quotas.NumOutstandingInvites, org.QuotaOutstandingInvites),
-			QuotaStorageLimitBytesPerDeployment: valOrDefault(plan.Quotas.StorageLimitBytesPerDeployment, org.QuotaStorageLimitBytesPerDeployment),
-			BillingCustomerID:                   org.BillingCustomerID,
-			PaymentCustomerID:                   org.PaymentCustomerID,
-		})
+		// create new subscription
+		_, err = s.admin.Biller.CreateSubscription(ctx, org.BillingCustomerID, plan)
 		if err != nil {
 			return nil, status.Error(codes.Internal, err.Error())
 		}
 	}
 
+	org, err = s.admin.DB.UpdateOrganization(ctx, org.ID, &database.UpdateOrganizationOptions{
+		Name:                                org.Name,
+		Description:                         org.Description,
+		QuotaProjects:                       valOrDefault(plan.Quotas.NumProjects, org.QuotaProjects),
```

Comment:
I am thinking that plan quotas should always have a value, if we forgot to set the metadata key for a quota in Orb then we use the default values 
```
const (
	DefaultQuotaProjects                       = 5
	DefaultQuotaDeployments                    = 10
	DefaultQuotaSlotsTotal                     = 20
	DefaultQuotaSlotsPerDeployment             = 5
	DefaultQuotaOutstandingInvites             = 200
	DefaultQuotaSingleuserOrgs                 = 3
	DefaultQuotaStorageLimitBytesPerDeployment = int64(5368709120)
)
```